python/tests/tornado/test_hyperbahn.py

```diff
@@ -57,8 +58,21 @@ def test_request():
 
 
 @pytest.mark.gen_test
-def test_register():
+def test_advertise(tchannel_server):
+    endpoint = b'ad'
+
+    f = Future()
+
+    @tornado.gen.coroutine
+    def execution(request, response):
+        body = yield request.get_body()
+        f.set_result(body)
+
+    tchannel_server.expect_call(endpoint, 'json').execute = execution
```

Comment:
What's the problem with using a TChannel object instead of this `tchannel_server` fixture? As said before, it's a leftover from when we didn't have a convenient interface to register endpoints on the TChannel. We should work towards eliminating its use from the codebase.

Check the old comment on this file for what I'm saying. The code will look about the same but you'll no longer be relying on this unclean assign-to-`execute` approach.